modules/space/shaders/fluxnodes_vs.glsl

```diff
@@ -0,0 +1,366 @@
+uniform float   perspectiveDistanceFactor;
+
+uniform float   maxNodeSize;
+uniform float   minNodeSize;
+uniform bool    usingPulse;
+
+// Inputs
+// Should be provided in meters VaPosition
+layout(location = 0) in vec3 in_position;
+
+// The extra value used to color lines. Location must correspond to VaColor in 
+// renderablefluxnodes.h
+layout(location = 1) in float fluxValue;
+
+// The extra value used to mask out parts of lines. Location must correspond to 
+// VaFiltering in renderablefluxnodes.h
+layout(location = 2) in float rValue;
+
+// These should correspond to the enum 'ColorMode' in renderablefluxnodes.cpp
+const int colorByFluxValue  = 0;
+const int uniformColor     = 1;
+
+const int uniformskip = 0;
+const int fluxSkip = 1;
+const int radiusSkip = 2;
+
+const int fluxMode = 0;
+const int RFlux = 1;
+const int R2Flux = 2;
+const int log10RFlux = 3;
+const int lnRFlux = 4;
+
+const int sizeScaling = 0;
+const int colorTables = 1;
+const int sizeAndColor = 2;
+const int illuminance = 3;
+
+const float AUtoMeter = 149597871000.0;
+varying out vec4    vs_color;
+varying out float   vs_depth;
+varying out vec2    vs_st;
+varying out float   camera_IsCloseEnough;
+varying out float   vs_closeToEarth;
+varying out double  vs_time;
+varying out vec3    vs_camerapos;
+
+vec4 getTransferFunctionColor(sampler1D InColorTable) {
+    // Remap the color scalar to a [0,1] range
+    float scaleValue = 0;
+    if(scalingMode == fluxMode){
+        scaleValue = fluxValue;
+    }
+    else if(scalingMode == RFlux){
+       scaleValue = rValue * fluxValue;
+    }
+    else if(scalingMode == log10RFlux){
+        //conversion from logbase e to log10 since glsl does not support log10. 
+        float logtoTen = log(rValue) / log(10);
+        scaleValue = logtoTen * fluxValue;
+    }
+    else if(scalingMode == lnRFlux){
+        scaleValue = log(rValue) * fluxValue;
+    }
+    else if(scalingMode == R2Flux){
+        scaleValue = rValue * rValue * fluxValue;
+    }
+
+    float lookUpVal = (scaleValue - colorTableRange.x)
+        /(colorTableRange.y - colorTableRange.x);
+    return texture(InColorTable, lookUpVal);
+}
+
+bool CheckvertexIndex(){
+    int nodeIndex = gl_VertexID;
+
+    if(nodeSkipMethod == uniformskip){
+        if(mod(nodeIndex, nodeSkip) == 0){
+            return true;
+        }
+    }
+    else if(nodeSkipMethod == fluxSkip){
+        if(fluxValue > nodeSkipFluxThreshold && mod(nodeIndex, nodeSkip) == 0){
+            return true;
+        }
+        if(fluxValue < nodeSkipFluxThreshold && mod(nodeIndex, nodeSkipDefault) == 0){
+            return true;
+        }
+    }
+    else if(nodeSkipMethod == radiusSkip){
+        if(rValue < nodeSkipRadiusThreshold && mod(nodeIndex, nodeSkip) == 0){
+            return true;
+        }
+        if(rValue > nodeSkipRadiusThreshold && mod(nodeIndex, nodeSkipDefault) == 0){
+            return true;
+        }
+    }
+    return false;
+}
+//todo fix gl_VertexID
```

Comment:
Either the comment can be removed or it should be expanded to be made understandable